client/src/components/App/index.js

```diff
@@ -1,12 +1,94 @@
 import React from 'react';
+import {
+  BrowserRouter as Router,
+  Switch,
+  Route,
+  Redirect,
+} from 'react-router-dom';
+import axios from 'axios';
+
 import './index.css';
+import AppBar from '../common/AppBar';
+import Limitation from '../common/limitation';
 
-function App() {
-  return (
-    <div>
-      <h1>Welcome in Applicants System</h1>
-    </div>
-  );
-}
+export default class App extends React.Component {
+  state = {
+    auth: null,
+  };
+
+  componentDidMount() {
+    axios
+      .get('/api/v1/auth')
+      .then((result) => {
+        this.setState({ ...result.data });
+      })
+      .catch(() => {
+        this.setState({ auth: false });
+      });
+  }
 
-export default App;
+  signupHandler = () => {
+    this.setState({
+      auth: true,
+    });
+  };
+
+  logoutHandler = () => {
+    axios.get('/api/v1/logout').then(() => {
+      this.setState({
+        auth: false,
+      });
+    });
+  };
+
+  render() {
+    const { auth } = this.state;
+    return (
+      <div>
+        <AppBar />
+        <Router>
+          <Switch>
+            <Route path="/404" render={() => '404'} />
+            <Route path="/500" render={() => '500'} />
+            {auth === null ? (
```

Comment:
@ha77an Instead of having an `auth` prop that can be three values (null, false or true) sometimes it's clearer if you have two props: `loading` (false or true) and `auth` (false or true).  It's not a super important change to make but it can make your code easier to understand.